### ISSUE TITLE
Windows 10 machines not impacted by ransomware

### ISSUE THREAD

**Issue by mpearn** [kieranwalsh/PowerShell#2]
This blog states that Windows 10 machines are not impacted by this attack: https://blogs.technet.microsoft.com/mmpc/2017/05/12/wannacrypt-ransomware-worm-targets-out-of-date-systems/

I'd suggest including an updated operating system filter to exclude the computer objects from AD.

**Comment by kieranwalsh** [comment]
Good point - that has been done in the latest version.
Thanks for the tip.

**Comment by mattclements** [comment]
Does this actually not impact Windows 10? My understanding is that "a fully updated Windows 10 machine is not affected"

**Comment by mpearn** [comment]
It's unclear - "Customers running Windows 10 were not targeted by the attack today".  I don't know if they could be in the future.

**Comment by onlabs-fr** [comment]
Windows 10 are also targeted. Refer to March MS17010 [https://technet.microsoft.com/en-us/library/security/ms17-010.aspx](url)
Only Windows 10 Creator Update is not affected.

**Comment by mattzuba** [comment]
Agree with @onlabs-fr - Windows 10 *should* be included in the search.

**Comment by mpearn** [comment]
I don't necessarily agree - whilst the computers are in scope for MS17-010, Microsoft themselves have stated that Windows 10 aren't affected by the ransomware attack specifically.  They wouldn't state this publically if they weren't confident that it wouldn't be impacted by this ransomware.  I think all machines should have 100% of all security patches, but when you're dealing with an urgent patching issue, I think focussing on the older machines should be a priority.

**Comment by udpPigeons** [comment]
To split hairs, and to take Microsoft literally, they said that Windows 10 machines weren't *targeted*, not that they weren't *affected*. Taking that at face value could simply mean they were vulnerable, but the ransomware did not attack Windows 10. It would be far safer for everyone to assume that Windows 10 is vulnerable and they need updates from MS17-010 and/or any successors.

**Comment by onlabs-fr** [comment]
In fact everyone's right,  Windows 10 has the vulnerability but the exploit doesn't work on this (probably because W10 wasn't exist or bearly new at the moment of writing the exploit).
Source Microsoft Security  : https://www.microsoft.com/security/portal/threat/encyclopedia/Entry.aspx?Name=Ransom:Win32/WannaCrypt
Focus on older versions is good for the very first days but who knows the following of this story... In this script, if W10 is removed, people will come and say "hey xxxxcrypt affects W10, how Can I handle this" ?
But I totally understand @mpearn

**Comment by mpearn** [comment]
Agreed.  Windows 10 should be reviewed as well.  I think for now, get the script working for the old versions as an absolute priority then later today work to get Windows 10 'back in'.  My concerns with the script currently are that there are least 6 Windows 10 SKUs, and looking at the 'superseded by' list of KBs, there's an exhaustive number of KBs for Windows 10 that aren't in the script currently.  I'd suggest taking Windows 10 out of the report until all KBs are in the script.  For example, on my Windows 10 v1511 64-bit SKU, I had to install KB4019473 which was the latest cumulative update for that SKU.  Only the previous version KB is in the script currently.  See my comment on the 'incorrect list of patches' issue list.

**Comment by udpPigeons** [comment]
I agree, fix the bigger holes first because they can get you in hot water quick, but obviously don't neglect other vulnerabilities. As @onlabs-fr said, I suspect this won't be the last we've heard of WannaCry our its variants.
